x/foundation/keeper/internal/migrations/v2/store.go

```diff
@@ -0,0 +1,39 @@
+package v2
+
+import (
+	"fmt"
+
+	"github.com/Finschia/finschia-sdk/codec"
+	storetypes "github.com/Finschia/finschia-sdk/store/types"
+	sdk "github.com/Finschia/finschia-sdk/types"
+	"github.com/Finschia/finschia-sdk/x/foundation"
+)
+
+// MigrateStore performs in-place store migrations from v1 to v2.
+func MigrateStore(ctx sdk.Context, storeKey storetypes.StoreKey, cdc codec.BinaryCodec, subspace Subspace) error {
+	store := ctx.KVStore(storeKey)
+
+	// migrate params
+	if err := migrateParams(ctx, store, cdc, subspace); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func migrateParams(ctx sdk.Context, store storetypes.KVStore, cdc codec.BinaryCodec, subspace Subspace) error {
+	bz := store.Get(ParamsKey)
+	if bz == nil {
+		return fmt.Errorf("params not found")
+	}
+	store.Delete(ParamsKey)
```

Comment:
How about moving after executing `subspace.SetParamSet()`, because there are error cases between `store.Delete()` and `SetParamsSet`.